packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -103,6 +112,24 @@ export function unstable_httpSubscriptionLink<
           >({
             from: eventSource,
             deserialize: transformer.output.deserialize,
+            tryHandleError: async (ev) => {
+              if (
```

Comment:
we don't offer the chance to reboot when the network error itself occurs and that event doesn't come with a status code, but when it recovers and there's an issue like 401 we do call this. So it can take an extra round trip to recover but feels easier to work with

Reply:
🤔 

This feels a bit off to me but I don't hate it either... 

I feel like `shouldReinitialize()` should be called whenever it's closed while there is an active subscription

Reply:
Sketching

```ts
// closed unexpectedly 
type Closed = { type: "closed" }
// errors from the original EventSource
type Failure = { type: "event", event: Event; }

// if it's an error in i.e. the URL callback or the eventSourceOptions-callback 
type CallbackError = { type: "error", error: TRPCClientError }

type Failure = CallbackError | Closed | Failure
type ShouldReinitialize = (opts: Failure) => MaybePromise<boolean>;

```

Reply:
I like the idea of that, I would also add a dedicated HttpError to that with status etc expanded because Event lacks the type safety here, but exposing the other states does make sense

Reply:
Right so I checked if there was any "Closed" type error via the test and doesn't seem to be, it may well fire on network disconnection but I've wrapped at "raw" and "http-error" types because I don't want to make too many assumptions about what non-http-error cases will look like but pulling out the status automatically makes a lot of sense as it's a common case